core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,116 +288,3206 @@ def validate(cls, item):
+        }
+
+    @classmethod
+    def _validate_contributors_summary(cls, item):
+        """Validate that contributor ids match the contributor ids obtained
+        from contributors summary.
+
+        Args:
+            item: CollectionSummaryModel to validate.
+        """
+        contributor_ids_from_contributors_summary = (
+            item.contributors_summary.keys())
+        if sorted(item.contributor_ids) != sorted(
+                contributor_ids_from_contributors_summary):
+            cls.errors['contributors summary check'] = (
+                'Model id %s: Contributor ids: %s do not match the contributor '
+                'ids obtained using contributors summary: %s') % (
+                    item.id, (',').join(sorted(item.contributor_ids)),
+                    (',').join(
+                        sorted(contributor_ids_from_contributors_summary)))
+
+    @classmethod
+    def _validate_language_code(cls, item):
+        """Validate that language code is present in allowed language codes.
+
+        Args:
+            item: CollectionSummaryModel to validate.
+        """
+        allowed_language_codes = [
+            language_item['code'] for language_item in (
+                constants.ALL_LANGUAGE_CODES)]
+
+        if item.language_code not in allowed_language_codes:
+            cls.errors['language code check'] = (
+                'Model id %s: Language code %s for model is unsupported' % (
+                    item.id, item.language_code))
+
+    @classmethod
+    def _validate_node_count(cls, item):
+        """Validate that node_count of model is equal to number of nodes
+        collection_contents in collection model.
+
+        Args:
+            item: CollectionSummaryModel to validate.
+        """
+        _, collection_model_tuples = cls.external_models[
+            'collection_model']
+        collection_model = collection_model_tuples[0][1]
+        nodes = collection_model.collection_contents['nodes']
+        if item.node_count != len(nodes):
+            cls.errors['node count check'] = (
+                'Model id %s: Node count: %s does not match the number of '
+                'nodes in collection_contents dict: %s') % (
+                    item.id, item.node_count, len(nodes))
+
+    @classmethod
+    def _validate_related_model_properties(cls, item):
+        """Validate that model properties match the corresponding collection
+        model and collection rights model properties.
+
+        Args:
+            item: CollectionSummaryModel to validate.
+        """
+        _, collection_model_tuples = cls.external_models[
+            'collection_model']
+        _, collection_rights_model_tuples = cls.external_models[
+            'collection_rights_model']
+        collection_model = collection_model_tuples[0][1]
+        collection_rights_model = collection_rights_model_tuples[0][1]
+
+        if item.title != collection_model.title:
```

Comment:
Create a list of properties to match and then test the match in a loop (similar to what you have done in tests)

Reply:
Done.